src/include/execution/sql/join_hash_table.h

```diff
@@ -272,4 +274,79 @@ inline HashTableEntryIterator JoinHashTable::Lookup<true>(const hash_t hash) con
   return HashTableEntryIterator(entry, hash);
 }
 
+//===----------------------------------------------------------------------===//
+//
+// Join Hash Table Iterator
+//
+//===----------------------------------------------------------------------===//
+/**
+ * A tuple-at-a-time iterator over the contents of a join hash table. The join hash table must be
+ * fully built either through a serial call to JoinHashTable::Build() or merged (in parallel) from
+ * other join hash tables through JoinHashTable::MergeParallel().
+ *
+ * Users use the OOP-ish iteration API:
+ * for (JoinHashTableIterator iter(table); iter.HasNext(); iter.Next()) {
+ *   auto row = iter.GetCurrentRowAs<MyFancyAssType>();
+ *   ...
+ * }
+ *
+ * // TODO(pmenon): Vectorized version, too.
+ */
+class HashTableNaiveIterator {
```

Comment:
Why is this the "naive" iterator? The comment above says that it is the "Join Hash Table Iterator"?

Reply:
I agree naming is an issue. There's two different iterators for the join hash table. One iterator takes a hash and lets you iterate over all the matching hash values called HashTableEntryIterator. This iterator simply iterates over all the entries in the hash table. I didn't know what to call it, because it is also a entry iterator.